### PR TITLE
refactor(behavior_path_planner): shift pull over

### PR DESCRIPTION
**Pull request by kosuke55**
Signed-off-by: kosuke55 <kosuke.tnp@gmail.com>

## Description

refactor shift pull over
- make functions
- rename variable
etc..

## Related links

[TIEIV INTERNAL LINK](https://tier4.atlassian.net/browse/T4PB-22423?searchSessionId=02707f5f-cb21-4836-a022-b80e42ee822a&searchObjectId=67836&searchContainerId=10206&searchContentType=issue)

## Tests performed

psim

tier4 interval scenario test
https://evaluation.tier4.jp/evaluation/reports/af547b2b-a999-5285-9edd-86e2ad418cf3?project_id=prd_jt
23/23

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/pull_over/shift_pull_over.hpp [hunk 1]
```diff
@@ -36,25 +35,32 @@ class ShiftPullOver : public PullOverPlannerBase
   ShiftPullOver(
     rclcpp::Node & node, const PullOverParameters & parameters,
     const LaneDepartureChecker & lane_departure_checker,
-    const std::shared_ptr<OccupancyGridBasedCollisionDetector> occupancy_grid_map);
+    const std::shared_ptr<OccupancyGridBasedCollisionDetector> & occupancy_grid_map);
```
Comment (rej55) [inline planning/behavior_path_planner/include/behavior_path_planner/scene_module/pull_over/shift_pull_over.hpp:38]: Is it okay to use reference value of shared pointer?

File: planning/behavior_path_planner/src/scene_module/pull_over/shift_pull_over.cpp [hunk 2]
```diff
@@ -38,321 +38,286 @@ ShiftPullOver::ShiftPullOver(
...
-    available_paths.push_back(path);
+  PullOverPath pull_over_path{};
+  pull_over_path.path =
+    pull_over_utils::combineReferencePath(road_lane_reference_path, shifted_path.path);
+  // shift path is connected to one, so partial_paths have only one
+  pull_over_path.partial_paths.push_back(pull_over_path.path);
+  pull_over_path.start_pose = path_shifter.getShiftLines().front().start;
+  pull_over_path.end_pose = path_shifter.getShiftLines().front().end;
+
+  // check enogh distance
```
Comment (rej55) [inline planning/behavior_path_planner/src/scene_module/pull_over/shift_pull_over.cpp:245]:
```suggestion
  // check enough distance
```

**Review by rej55 — COMMENTED** [review]
Thank you for your contribution.
I added some comments.
Could you check them?

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/pull_over/shift_pull_over.hpp (hunk 1 above)
Comment (kosuke55) [reply planning/behavior_path_planner/include/behavior_path_planner/scene_module/pull_over/shift_pull_over.hpp:38]: I think it is not needed, but clang-tidy suggests this, so I followed it.

**Review by rej55 — APPROVED** [review]

**Pull request merged**